src/components/Tokens/TokenDetails/TokenDetail.tsx

```diff
@@ -199,12 +206,22 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
     twitterName,
   }))(tokenDetailData)
 
-  if (!token || !token.name || !token.symbol) {
+  if (!token || !token.name || !token.symbol || !connectedChainId) {
     return <LoadingTokenDetail />
   }
 
-  const tokenName = tokenDetailData.name
-  const tokenSymbol = tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol
+  const wrappedNativeCurrency = WRAPPED_NATIVE_CURRENCY[connectedChainId]
+  const isWrappedNativeToken = wrappedNativeCurrency && wrappedNativeCurrency.address === token.address
+
+  if (isWrappedNativeToken) {
+    currency = nativeOnChain(connectedChainId)
+  }
+
+  const tokenName = isWrappedNativeToken && currency ? currency.name : tokenDetailData.name
+  const tokenSymbol =
```

Comment:
this is getting really hard to follow... why can't you use the tokenDetailData stuff anymore?

Reply:
I think they still want the wETH tokenDetail stuff but the ETH symbols, name, and logo

Reply:
oh boy sounds good